### DISCUSSION TITLE
Model has been downloaded but the SHA256 checksum does not not match. Please retry loading the model.

### DISCUSSION THREAD

**Discussion by WorkAlbertX** [openai/whisper#1027] [Q&A] [answered]
I tried to run Whisper on command line with Python3.10.9,PyTorch for CUDA11.7 and CUDA11.7,but every time I run it shows "Model has been downloaded but the SHA256 checksum does not not match. Please retry loading the model" and redownload the pt file.It bothered me for a long time,is there an enthusiastic person who can help me?Appreciate.

Total error message:
PS D:\Test_for_Whisper> whisper audio.mp3
C:\Users\Albert_Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\whisper\__init__.py:48: UserWarning: C:\Users\Albert_Administrator\.cache\whisper\small.pt exists, but the SHA256 checksum does not match; re-downloading the file
  warnings.warn(f"{download_target} exists, but the SHA256 checksum does not match; re-downloading the file")
 99%|██████████████████████████████████████▊| 459M/461M [00:55<00:00, 8.65MiB/s]
Traceback (most recent call last):
  File "C:\Users\Albert_Administrator\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Albert_Administrator\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\Albert_Administrator\AppData\Local\Programs\Python\Python310\Scripts\whisper.exe\__main__.py", line 7, in <module>
  File "C:\Users\Albert_Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\whisper\transcribe.py", line 310, in cli
    model = load_model(model_name, device=device, download_root=model_dir)
  File "C:\Users\Albert_Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\whisper\__init__.py", line 108, in load_model
    checkpoint_file = _download(_MODELS[name], download_root, in_memory)
  File "C:\Users\Albert_Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\whisper\__init__.py", line 62, in _download
    raise RuntimeError("Model has been downloaded but the SHA256 checksum does not not match. Please retry loading the model.")
RuntimeError: Model has been downloaded but the SHA256 checksum does not not match. Please retry loading the model.

**Comment by WorkAlbertX** [comment, marked as answer]
I dont kown what happened,but the error has been fixed automatically after restarting.

**Reply by EtienneAb3d** [reply]
I had the same problem on an unstable machine with a lot of disk read/write errors. After a (very) long search, I found this was due to a SWAP configured on a SSD disk (I didn't have this problem before with some other SSD).

**Reply by adhiraj2001** [reply]
@EtienneAb3d I seem to have the same problem as you, what workaround did you use for this ?

**Reply by NIXON-L** [reply]
My problem was resolved by removing the ~/.cache/torch/whisperx-vad-segmentation.bin file

**Reply by jovanovic-djo** [reply]
True, I had the same issue, but after re-running it a couple of times it worked, so..

**Comment by Yongle-Fu** [comment]
same issue

**Comment by zhuiyi6** [comment]
C:\Users\Administrator.DESKTOP-DHKFNAB\Desktop>whisper test.mp3 --model large-v2 --language Chinese --output_format srt
C:\Users\Administrator.DESKTOP-DHKFNAB\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\__init__.py:63: UserWarning: C:\Users\Administrator.DESKTOP-DHKFNAB\.cache\whisper\large-v2.pt exists, but the SHA256 checksum does not match; re-downloading the file
  warnings.warn(
 15%|█████▍                               | 435M/2.87G [59:59<5:46:02, 127kiB/s]
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\Administrator.DESKTOP-DHKFNAB\AppData\Local\Programs\Python\Python311\Scripts\whisper.exe\__main__.py", line 7, in <module>
  File "C:\Users\Administrator.DESKTOP-DHKFNAB\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\transcribe.py", line 440, in cli
    model = load_model(model_name, device=device, download_root=model_dir)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Administrator.DESKTOP-DHKFNAB\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\__init__.py", line 131, in load_model
    checkpoint_file = _download(_MODELS[name], download_root, in_memory)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Administrator.DESKTOP-DHKFNAB\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\__init__.py", line 85, in _download
    raise RuntimeError(
RuntimeError: Model has been downloaded but the SHA256 checksum does not not match. Please retry loading the model.

how to solve this problem

**Comment by Bailey-24** [comment]
restart what?

**Comment by Slepetys** [comment]
I solve it under WINDOWS 11 miniconda environment deleting the whole huggingface directory under default user/.cache. 
Was a kind of brute force, implying that I needed to redownload many hugging face models, but it worked.

**Reply by labtec901** [reply]
This worked for me, thanks.

**Comment by search620** [comment]
remove those files and run the script again
"C:\Users\\.cache\torch\whisperx-vad-segmentation.bin"
"C:\Users\USER-NAME\.cache\torch\whisperx-vad-segmentation.bak"

**Reply by fraschm1998** [reply]
This fixed it for me on linux

**Reply by sdekna** [reply]
I had the same problem, my device is linux based, so `rm ~/.cache/torch/whisperx-vad-segmentation.bin` worked for me.
Thanks for the help.

**Reply by QuietWinterX** [reply]
this

**Reply by kovogo** [reply]
it works for me

**Comment by sirEgghead** [comment]
I have this same issue.  I am running on Ubuntu 22.04.  Fresh Whisper install.  Have not yet had a successful transcribe.  Download of the appropriate .pt file gets to 100% then sits there.  I left one for 48 hours.  Upon restarting Whisper, it says that the SHA checksum does not match and will download again.  Same results.  Tried manually deleting the .pt files from the .cache.  No change in result.

**Reply by misutoneko** [reply]
Hi,

A late answer for sure, but here are some misc ideas for Linux...

The models are located in ~/.cache/whisper by default.
You could take a look in there and even check the SHA sums manually, if need be.
The tiny models are about ~73MB in size so should be pretty fast to dl unless you're on dialup :D

It's best to start out with tiny.en or other small models first.
Otherwise you could be in for a looong wait.
Besides, you should see good results with the jfk sample, even with the tiny models.

You could also try running it with CPU only by setting the env variable CUDA_VISIBLE_DEVICES=-1
That would be slow, but it'll tell you if your problem is CUDA-related.

GL

**Comment by royallenppt** [comment]
it seems related to network as well. i tried 3 times, and it can work

**Comment by lemo366** [comment]
我遇到了同样的问题，我用的google colab尝试了下面的两个命令就可以顺利运行了
! rm -rf ~/.cache/whisper
! rm -rf ~/.cache/torch

**Comment by medford-transcripts** [comment]
FYI, this happened to me, and I know what caused it. While transcribing, I got this error message:

Lightning automatically upgraded your loaded checkpoint from v1.5.4 to v2.4.0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint %USERPROFILE%\.cache\torch\whisperx-vad-segmentation.bin`

I issued the command, but I was using a CPU, not GPU, so I added the --map-to-cpu option:

python -m pytorch_lightning.utilities.upgrade_checkpoint --map-to-cpu %USERPROFILE%\.cache\torch\whisperx-vad-segmentation.bin

But then I got this error. I fixed it by deleting %USERPROFILE%\.cache\torch\whisperx-vad-segmentation.bin, and just accepting the original error message.

**Comment by NIXON-L** [comment]
It seems that my problem was resolved by removing the ~/.cache/torch/whisperx-vad-segmentation.bin file

**Comment by huyueyue17** [comment]
same to me, but my system is Mac OS, how can I fix it?

**Comment by MarcNBarrett** [comment]
```
rm ~/.cache/torch/whisperx-vad-segmentation.bin
rm: cannot remove '/home/username/.cache/torch/whisperx-vad-segmentation.bin': No such file or directory

```

I am on a Linux system (Ubuntu 24.04).

I have tried everything. I even deleted the entire whisper directory in .cache.

**Reply by sawyerbutton** [reply]
Here is what i did and fix this issue,
I clear the all .pt files in ~/.cache/whisper folder.
Download original model pt file like base pt from `https://openaipublic.azureedge.net/main/whisper/models/ed3a0b6b1c0edf879ad9b11b1af5a0e6ab5db9205f891f668f8b0e6c6326e34e/base.pt`
and then it works.

**Comment by bbbjames** [comment]
hi! i too had this issue, i'm on windows atm

looking at the script download sha256, finding the link in init.py and downloading via my browser, they dont match 🤷

<img width="722" height="374" alt="image" src="https://github.com/user-attachments/assets/e99973e0-b4c7-4218-b614-20a6b169d96e" />

i replaced the script downloaded version in .cache\whisper with the browser downloaded version; 

anyway it works now...

```bash
Detected language: English
[00:00.000 --> 00:05.000]  Hi, I'm Bob, and this is the audio transcription test
[00:05.600 --> 00:06.900]  for Whisper AI.
```

bit concerned, can anyone help me understand why the hashes don't match please?

**Comment by lemo366** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
